mdoc-js-worker/src/main/scala/ScalaJSWorker.scala

```diff
@@ -0,0 +1,92 @@
+package mdoc.js.worker;
+
+import mdoc.js.api._
+import java.nio.file.Path
+import org.scalajs.linker.MemOutputDirectory
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.scalajs.linker.StandardImpl
+import org.scalajs.linker.interface.StandardConfig
+import org.scalajs.linker.interface.ModuleKind
+import ModuleType._
+import org.scalajs.linker.PathIRContainer
+import org.scalajs.linker.interface.IRFile
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import java.{util => ju}
+import org.scalajs.logging.Logger
+import org.scalajs.logging.Level
+import org.scalajs.linker.standard.MemIRFileImpl
+import org.scalajs.linker.interface.Semantics
+
+class ScalaJSWorker(
+    config: ScalajsConfig,
+    logger: Logger
+) extends ScalajsWorkerApi {
+  case class IFile(mem: IRFile) extends ScalajsWorkerApi.IRFile
+
+  val linker = {
+    val cfg =
+      StandardConfig()
+        .withSemantics {
+          if (config.fullOpt) Semantics.Defaults.optimized
+          else Semantics.Defaults
+        }
+        .withBatchMode(config.batchMode)
+        .withClosureCompilerIfAvailable(config.closureCompiler)
+        .withSourceMap(config.sourceMap)
+        .withModuleKind {
+          config.moduleType match {
+            case NoModule => ModuleKind.NoModule
+            case ESModule => ModuleKind.ESModule
+            case CommonJSModule => ModuleKind.CommonJSModule
+          }
+        }
+    StandardImpl.clearableLinker(cfg)
+  }
+
+  var cachedFiles = Seq.empty[org.scalajs.linker.interface.IRFile]
+
+  val cache = StandardImpl.irFileCache().newCache
+
+  override def cache(x: Array[Path]): Unit =
+    cachedFiles = Await.result(
+      PathIRContainer
+        .fromClasspath(x.toSeq)
+        .map(_._1)
+        .flatMap(cache.cached),
+      Duration.Inf
```

Comment:
To be honest the entire JsModifier process is very synchronous and futures are always waited on, even in the current main branch.

We can always change the interface as it is internal, so I wanted to keep it simple and avoid introducing potentially configurable timeouts because its brittle.

Reply:
Ok sure!